### PR TITLE
Edited dlgRank

### PR DESCRIPTION
**Pull request by maxwellfundi**

### CODE REVIEW

File: instat/dlgRank.vb [hunk 1]
```diff
@@ -21,25 +21,28 @@ Public Class dlgRank
...
+    Private Sub InitialiseDialog()
+        ucrBase.clsRsyntax.SetFunction("rank")
+        ucrReceiverRank.Selector = ucrSelectorForRank
+        ucrReceiverRank.SetDataType("numeric")
+        ucrBase.iHelpTopicID = 25
+        ucrInputColName.SetPrefix("Rank")
+        ucrInputColName.SetItemsTypeAsColumns()
+        ucrInputColName.SetDefaultTypeAsColumn()
+        ucrInputColName.SetDataFrameSelector(ucrSelectorForRank.ucrAvailableDataFrames)
+        ucrBase.clsRsyntax.SetAssignTo(strAssignToName:=ucrInputColName.GetText, strTempDataframe:=ucrSelectorForRank.ucrAvailableDataFrames.cboAvailableDataFrames.Text, strTempColumn:=ucrInputColName.GetText)
```
Comment (dannyparsons) [inline instat/dlgRank.vb]: Again, you shouldn't need this here now that you have the NameChanged event. See if it works if this is removed. Any problems, let me know.

**Comment by maxwellfundi** [comment]
@DannyParsons i have now corrected this and looks good.

File: instat/dlgRank.vb [hunk 2]
```diff
@@ -20,26 +20,28 @@ Public Class dlgRank
     Public bFirstLoad As Boolean = True
 
     Private Sub dlgRank_Load(sender As Object, e As EventArgs) Handles MyBase.Load
+        ucrInputColName.SetPrefix("Rank")
+        ucrInputColName.SetItemsTypeAsColumns()
+        ucrInputColName.SetDefaultTypeAsColumn()
+        ucrInputColName.SetDataFrameSelector(ucrSelectorForRank.ucrAvailableDataFrames)
```
Comment (dannyparsons) [inline instat/dlgRank.vb]: These things are constant so they should also go in InitialiseDialog() as well.

**Comment by maxwellfundi** [comment]
@dannyparsons  I was wondering if this is ready to be merged or what more should i have to do on it

File: instat/dlgRank.vb [hunk 3]
```diff
@@ -52,25 +55,25 @@ Public Class dlgRank
...
         ucrSelectorForRank.Reset()
+        ucrInputColName.SetPrefix("Rank")
         TestOKEnabled()
     End Sub
 
     Private Sub ReopenDialog()
-
         SetTiesValues()
         setMissingValue()
+        ucrInputColName.SetPrefix("Rank")
```
Comment (dannyparsons) [inline instat/dlgRank.vb]: These three lines can be removed, and instead we can set default values in the set defaults

File: instat/dlgRank.vb [hunk 4]
```diff
@@ -52,25 +55,25 @@ Public Class dlgRank
 
     ' Sub that runs only the first time the dialog loads
     Private Sub SetDefaults()
-        ucrReceiverRank.SetMeAsReceiver()
         ucrReceiverRank.Selector = ucrSelectorForRank
+        ucrReceiverRank.SetMeAsReceiver()
         rdoKeptAsMissing.Checked = True
         rdoAverage.Checked = True
         ucrSelectorForRank.Reset()
+        ucrInputColName.SetPrefix("Rank")
```
Comment (dannyparsons) [inline instat/dlgRank.vb]: Here set "Average" as the default for "Ties", and "Kept as missing" as the default for "Missing values"

**Comment by maxwellfundi** [comment]
@dannyparsons This is now done

**Pull request merged**